src/libfsm/determinise.c

```diff
@@ -252,31 +256,57 @@ fsm_determinise(struct fsm *nfa)
 		fsm_move(nfa, dfa);
 	}
 
+#if EXPENSIVE_CHECKS
+	assert(fsm_all(nfa, fsm_isdfa));
```

Comment:
`fsm_minimise` also checks this, but it makes sense to check here since it if it fails, it could be a bug in `fsm_determinise` rather than API misuse.